packages/tools/src/tools/base/BaseTool.ts

```diff
@@ -83,7 +83,35 @@ abstract class BaseTool implements IBaseTool {
     operationData: unknown
   ): any {
     const { strategies, activeStrategy } = this.configuration;
-    return strategies[activeStrategy].call(this, enabledElement, operationData);
+    return strategies[activeStrategy]?.call(
+      this,
+      enabledElement,
+      operationData
+    );
+  }
+
+  /**
+   * It applies the active strategy down initialization, if any.
+   * @param enabledElement - The element that is being operated on.
+   * @param operationData - The data that needs to be passed to the strategy.
+   * @returns The result of the strategy.
+   */
+  public applyActiveStrategyEvent(
+    enabledElement: Types.IEnabledElement,
+    operationData: unknown,
+    eventType:
+      | 'initDown'
+      | 'completeUp'
+      | 'rejectPreview'
+      | 'acceptPreview'
+      | 'preview'
+  ): any {
+    const { strategies, activeStrategy } = this.configuration;
+    return strategies[activeStrategy][eventType]?.call(
+      this,
+      enabledElement,
+      operationData
+    );
   }
```

Comment:
I'm not sure what is this, the name does not suggest what it does nor the documentaiton, what do you mean by `It applies the active strategy down initialization, if any.`